code/modules/mob/living/carbon/xenomorph/damage_procs.dm

```diff
@@ -169,27 +169,31 @@
 	} \
 
 /mob/living/carbon/xenomorph/adjustBruteLoss(amount, updating_health = FALSE)
-
 	var/list/amount_mod = list()
 	SEND_SIGNAL(src, COMSIG_XENOMORPH_BRUTE_DAMAGE, amount, amount_mod)
 	for(var/i in amount_mod)
 		amount -= i
 
+	adjustBruteLossDirect(amount, updating_health)
+
+///Adjusts bruteloss without sending a signal. Useful when it can cause a signal loop
+/mob/living/carbon/xenomorph/proc/adjustBruteLossDirect(amount, updating_health = FALSE)
 	HANDLE_OVERHEAL(amount)
 
 	bruteloss = max(bruteloss + amount, 0)
 
 	if(updating_health)
 		updatehealth()
 
-
 /mob/living/carbon/xenomorph/adjustFireLoss(amount, updating_health = FALSE)
-
 	var/list/amount_mod = list()
 	SEND_SIGNAL(src, COMSIG_XENOMORPH_BURN_DAMAGE, amount, amount_mod)
 	for(var/i in amount_mod)
 		amount -= i
+	adjustFireLossDirect(amount, updating_health)
 
+///Adjusts fireloss without sending a signal. Useful when it can cause a signal loop
+/mob/living/carbon/xenomorph/proc/adjustFireLossDirect(amount, updating_health = FALSE)
```

Comment:
uuuh, what is this??

Reply:
It modifies fireloss without sending the signal, so the action doesn't trigger effects that activate on being damaged and such. Should I make it a define?

Reply:
that's extremly iffy. If you send a signal when an action is done, you do it all the time. You don't have one proc that does a signal and the other not, this is awful design